File: apps/admin_mobile/lib/application/use_cases/clients/update_client_use_case.dart
import '../../../domain/entities/client_resource.dart';
import '../../../domain/repositories/client_resource_repository.dart';

class UpdateClientUseCase {
  final ClientResourceRepository _repo;
  UpdateClientUseCase(this._repo);

  Future<ClientResource> call(
    int id, {
    Map<String, dynamic>? data,
    String? plate,
    String? brand,
    String? model,
    String? color,
    String? type,
  }) =>
      _repo.update(
        id,
        data: data,
        plate: plate,
        brand: brand,
        model: model,
        color: color,
        type: type,
      );
}

File: apps/admin-v2/src/application/use-cases/clients/update-client.use-case.ts
import type { ClientResourceRepository, CreateClientResourceData } from '@/domain/repositories/client-resource.repository';

export class UpdateClientUseCase {
  constructor(private repo: ClientResourceRepository) {}

  execute(id: string, data: Partial<CreateClientResourceData>) {
    return this.repo.update(id, data);
  }
}
